### ISSUE TITLE
(feat req) Add codemod suggestions for related updates.

### ISSUE THREAD

**Issue by hulkish** [renovateapp/renovate#1678]
#### This is a:

* [ ] Bug report (non-security related)
* [X] Feature request
* [ ] I'm not sure which of those it is

#### I'm using:

* [ ] The Renovate GitHub App
* [X] Self-hosted GitHub
* [ ] Self-hosted GitLab
* [ ] Self-hosted VSTS

#### Please describe the issue:

This is a feature request.

##### Concept

Typically for major version bumps, there will require some refactoring. It would be nice if renovate was somehow able to notice this and make suggestions for existing codemods to assist in the upgrade.

- Example tools to refer to in suggestions:
  - https://github.com/facebook/jscodeshift
  - https://github.com/reactjs/react-codemod
  - https://github.com/cpojer/js-codemod/
  - https://github.com/jhgg/js-transforms

An example a comment for a major version pr renovate creates might look like using this feature:

> (`react@^15.0.0` => `react@^16.0.0`
> Here are some codemods that may help you with this upgrade:
> - https://github.com/reactjs/react-codemod#react-proptypes-to-prop-types
> - https://github.com/reactjs/react-codemod#pure-component

**Comment by rarkins** [comment]
Good idea in theory but challenging in practice. How would Renovate detect the right ones to recommend? Unless the upstream dependencies start providing the codemods themselves that is.
